### PR TITLE
Publisher overview stepper and Left Menu revamp

### PR DESCRIPTION
**Pull request by Krishanx92**
Screenshot from 2021-03-22 10-46-07

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+
+
+const ColorlibConnector = withStyles({
+    alternativeLabel: {
+        top: 22,
+    },
+    active: {
+        '& $line': {
+            backgroundImage:
+                'linear-gradient(to left, #50BCEC 50%, #B1D31E 50%)',
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx:32]: Are we using this linear grading effect now ?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+import CloseIcon from '@material-ui/icons/Close';
+import { Link } from 'react-router-dom';
+import grey from '@material-ui/core/colors/grey';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Button from '@material-ui/core/Button';
+import AuthManager from 'AppData/AuthManager';
+import Typography from '@material-ui/core/Typography';
+
+
+const ColorlibConnector = withStyles({
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx:25]: is this a redundant classes HOC ? `useColorlibStepIconStyles` is having same ?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+        backgroundColor: grey[500],
+        zIndex: 1,
+        width: 15,
+        height: 15,
+        borderRadius: '50%',
+    },
+}));
+
+function getSteps() {
+    return ['Develop', 'Deploy', 'Test', 'Publish'];
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx:122]: Shall we move this to a simple constant ? not a function ?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+    const [api, updateAPI] = useAPI();
+    const [isUpdating, setUpdating] = useState(false);
+    const [deploymentsAvailable, setDeploymentsAvailable] = useState(false);
+    const [lifecycleState, setlifecycleState] = useState(null);
+    const isEndpointAvailable = api.endpointConfig !== null;
+    const isTierAvailable = api.policies.length !== 0;
+    const steps = getSteps();
+
+    let activeStep = 0;
+    if (api && (api.type === 'WEBSUB' || isEndpointAvailable) && isTierAvailable && !deploymentsAvailable) {
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx:136]: Shall we move this api types to api.js , ideally there should be a method such as `isWebSub` to check this type

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx [hunk 5]
```diff
@@ -0,0 +1,427 @@
...
+        activeStep = 3;
+    } else if (lifecycleState === 'Published' && api
+        && (isEndpointAvailable || api.type === 'WEBSUB') && isTierAvailable && deploymentsAvailable) {
+        activeStep = 4;
+    }
+
+    useEffect(() => {
+        api.getLcState(api.id)
+            .then((result) => {
+                setlifecycleState(result.body.state);
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx:151]: AFAIR current lifecycle state is already there in the API object ? so do we need to featch it again ?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx [hunk 6]
```diff
@@ -0,0 +1,427 @@
...
+
+    /**
+ * Update the LifeCycle state of the API
+ *
+ */
+    function updateLCStateOfAPI(apiId, state) {
+        setUpdating(true);
+        const promisedUpdate = api.updateLcState(apiId, state);
+        promisedUpdate
+            .then(() => {
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx:166]: We can simplify this as

```
then(updateAPI).then()

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx [hunk 7]
```diff
@@ -0,0 +1,427 @@
...
+                updateAPI()
+                    .then()
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx:168]: If we don't do anything with the asyn result we can simply omit this `then` catch doesn't need to be followed by a `then()` call

Comment (tmkasun) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx:168]: BTW shouldn't we notify the user about the success case?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx [hunk 8]
```diff
@@ -0,0 +1,427 @@
...
+                                                />
+                                            </Typography>
+                                        </Grid>
+                                    </Grid>
+                                    {api.type !== 'WEBSUB' && (
+                                        <Grid
+                                            container
+                                            direction='row'
+                                            alignItems='left'
+                                            style={{ marginLeft: '15px' }}
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx:220]: shall we wrap with a box and use `ml={2}` prop to do the alignments ?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx [hunk 9]
```diff
@@ -0,0 +1,427 @@
...
+                                                </Link>
+                                            </Grid>
+                                        </Grid>
+                                    )}
+                                    <Grid
+                                        container
+                                        direction='row'
+                                        alignItems='center'
+                                        style={{ marginLeft: '28px' }}
+                                        justify='center'
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx:254]: For margins and paddings pls use MUI box

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx [hunk 10]
```diff
@@ -0,0 +1,427 @@
...
+}
+
+NewCustomizedStepper.propTypes = {
+    classes: PropTypes.shape({}).isRequired,
+    api: PropTypes.shape({
+        id: PropTypes.string,
+    }).isRequired,
+};
+
+NewCustomizedStepper.contextType = ApiContext;
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx:427]: `contextType` is only valid for class components? 

This is a no-op actually 😀 

https://reactjs.org/docs/context.html#classcontexttype

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/index.jsx [hunk 11]
```diff
@@ -47,14 +46,22 @@ import Utils from 'AppData/Utils';
 import ResourceNotFound from 'AppComponents/Base/Errors/ResourceNotFound';
 import CustomIcon from 'AppComponents/Shared/CustomIcon';
 import LeftMenuItem from 'AppComponents/Shared/LeftMenuItem';
+
 import API from 'AppData/api';
 import APIProduct from 'AppData/APIProduct';
+
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/index.jsx:52]: Shall we remove this unnecessary white spaces

**Review by tmkasun — APPROVED** [review]

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx (hunk 1 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx:32]: This is only for the connector, I remove it from
Screenshot from 2021-03-23 11-56-26
 Circle view

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx (hunk 2 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx:25]: No two separate class, one for connector one for circle Step

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx (hunk 3 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx]: Fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx (hunk 5 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx]: Fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/index.jsx (hunk 11 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/index.jsx]: Fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx (hunk 10 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx]: Fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx (hunk 9 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx]: Fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx (hunk 8 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx]: Fixed

**Pull request merged**